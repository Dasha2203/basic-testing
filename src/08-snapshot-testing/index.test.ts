import { generateLinkedList } from './index';

const data = [
  {
    fruit: 'apple',
    id: 1,
  },
  {
    fruit: 'banana',
    id: 2,
  },
  {
    fruit: 'cherry',
    id: 3,
  },
];

const expectedData = {
  value: {
    fruit: 'apple',
    id: 1,
  },
  next: {
    value: {
      fruit: 'banana',
      id: 2,
    },
    next: {
      value: {
        fruit: 'cherry',
        id: 3,
      },
      next: {
        value: null,
        next: null,
      },
    },
  },
};

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    expect(generateLinkedList(data)).toStrictEqual(expectedData);
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    expect(generateLinkedList(data)).toMatchSnapshot();
  });
});
