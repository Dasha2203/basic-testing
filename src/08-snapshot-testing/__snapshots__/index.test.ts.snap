// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": {
        "fruit": "cherry",
        "id": 3,
      },
    },
    "value": {
      "fruit": "banana",
      "id": 2,
    },
  },
  "value": {
    "fruit": "apple",
    "id": 1,
  },
}
`;
